APMrover2/sensors.cpp

```diff
@@ -128,14 +128,17 @@ void Rover::update_wheel_encoder()
         const float delta_angle = curr_angle_rad - wheel_encoder_last_angle_rad[i];
         wheel_encoder_last_angle_rad[i] = curr_angle_rad;
 
+        // save cumulative distances at current time (in meters)
+        wheel_encoder_last_distance_m[i] = g2.wheel_encoder.get_distance(i);
+
         // calculate delta time
         float delta_time;
         const uint32_t latest_sensor_update_ms = g2.wheel_encoder.get_last_reading_ms(i);
-        const uint32_t sensor_diff_ms = latest_sensor_update_ms - wheel_encoder_last_update_ms[i];
+        const int32_t sensor_diff_ms = latest_sensor_update_ms - wheel_encoder_last_update_ms[i];
```

Comment:
why this change ? normally we shouldn't have negative value

Reply:
But we have - for example, when wheel is not moving. Then wheel_encoder.get_last_reading_ms(i) reports some old time stamp, while wheel_encoder_last_update_ms[i] is always updated with the last EKF update (wheel_encoder_last_ekf_update_ms).